aws/data_source_aws_vpc.go

```diff
@@ -141,6 +162,17 @@ func dataSourceAwsVpcRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("state", vpc.State)
 	d.Set("tags", tagsToMap(vpc.Tags))
 
+	cidrAssociations := []interface{}{}
+	for _, associationSet := range vpc.CidrBlockAssociationSet {
+		association := map[string]interface{}{
+			"association_id": aws.StringValue(associationSet.AssociationId),
+			"cidr_block":     aws.StringValue(associationSet.CidrBlock),
+			"state":          aws.StringValue(associationSet.CidrBlockState.State),
+		}
+		cidrAssociations = append(cidrAssociations, association)
+	}
+	d.Set("cidr_block_associations", cidrAssociations)
```

Comment:
When using `d.Set()` with aggregate types (TypeList, TypeSet, TypeMap), [we should perform error checking to prevent issues where the code is not properly able to set the Terraform state](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#error-checking-aggregate-types). e.g.

```go
if err := d.Set("cidr_block_associations", cidrAssociations); err != nil {
  return fmt.Errorf("error setting cidr_block_associations: %s", err)
}
```

Luckily this was working just fine before so will adjust this on merge. 👍